server/util/shlex/shlex.go

```diff
@@ -0,0 +1,64 @@
+// shlex contains facilities for shell code parsing and generation.
+
+package shlex
+
+import (
+	"regexp"
+	"strings"
+
+	gshlex "github.com/google/shlex"
+)
+
+var (
+	allSafeCharsRegexp   = regexp.MustCompile(`^[A-Za-z0-9/_\-]+$`)
+	flagAssignmentRegexp = regexp.MustCompile(`^--[A-Za-z_-]+=`)
+)
+
+// Split parses the given shell command and returns the canonical tokenized
+// arguments.
+//
+// Example:
+//
+//	shlex.Split("  foo --bar='/Quoted/Path/With Spaces'  ")
+//	// Returns: []string{"foo", "--bar=/Quoted/Path/With Spaces"}
+func Split(command string) ([]string, error) {
+	return gshlex.Split(command)
+}
+
+// Quote escapes the given shell tokens so that each argument would be treated
```

Comment:
Talked offline about this, but could you add more comments to the code and to the PR description giving context for why this is needed